express-post-json/server.js

```diff
@@ -2,21 +2,21 @@ import express from 'express';
 
 const app = express();
 
-const nextId = 1;
+let nextId = 1;
 
 const grades = {};
 
-app.get('/api/grades', (req, res, next) => {
+app.use(express.json());
+
+app.get('/api/grades', (req, res) => {
   res.json(Object.values(grades));
-  next();
 });
 
-app.use(express.json());
-
 app.post('/api/grades', (req, res) => {
   grades[nextId] = req.body;
   grades[nextId].id = nextId;
-  res.status(201).send(grades);
+  nextId++;
+  res.status(201).send(req.body);
```

Comment:
I believe this works since line 16 copies the reference and line 17 modifies the original object, but it's a bit confusing and takes some thought to figure out if the response will contain the ID. It would be more clear if you created a new object, added the ID, assigned that object to the `grades` array and returned it.